test/browser.test.js

```diff
@@ -39,3 +39,72 @@ test('accepts custom history object', t => {
 	const wrapper = render(<BrowserComponent history={history}/>);
 	t.true(wrapper.text().includes('42'));
 });
+
+test('allows access to specified path if authenticated', t => {
+	const {BrowserRouter, AuthenticatedRoute, history} = require('../index');
+	history.location.pathname = '/about';
+	const isAuthenticated = true;
+	const BrowserComponent = props => (
+		<BrowserRouter {...props}>
+			<div>
+				<AuthenticatedRoute
+					exact
+					path="/about"
+					isAuthenticated={isAuthenticated}
+					component={App}
+				/>
+				<Login/>
+			</div>
+		</BrowserRouter>
+	);
+
+	const wrapper = mount(<BrowserComponent history={history}/>);
+
+	t.true(wrapper.text().includes('ABOUT'));
+});
+
+test('redirects to /login if not authenticated', t => {
+	const {BrowserRouter, AuthenticatedRoute, history} = require('../index');
+	history.location.pathname = '/test';
+	const isAuthenticated = false;
+	const BrowserComponent = props => (
+		<BrowserRouter {...props}>
+			<div>
+				<AuthenticatedRoute
+					exact
+					path="/"
+					isAuthenticated={isAuthenticated}
+					component={App}
+				/>
+				<Login/>
+			</div>
+		</BrowserRouter>
+	);
+
+	const wrapper = mount(<BrowserComponent history={history}/>);
+
+	t.true(wrapper.text().includes('LOGIN'));
+});
+
+test('AuthenticatedRoute works for nested routes', t => {
```

Comment:
I'm not sure if this test is testing nested routes, but I might be wrong. Anyway, from the `react-router` docs, nested routes is when you use a `<Route>` inside a `<Route>`. You may check examples here: https://reacttraining.com/react-router/web/guides/quick-start/2nd-example-nested-routing

Reply:
I actually wanted to confirm that with the project owner, since my first thought was of nested routes as in **/route1/nested1**, which matchPath solves by breaking it into path (route1) and parameters (nested1), but it might actually be a combination of both since AuthenticatedRoute does have an example with nested routing inside, I'll give it a try

Reply:
> but it might actually be a combination of both since AuthenticatedRoute does have an example with nested routing inside

Yup, a combination of both.

Reply:
Sure! I'll get back to it whenever I have some time 😄

Reply:
I just pushed a commit with two tests that validate nested routing with nested routes, let me know if I'm missing something or if there's anything you'd like me to change

Reply:
Looks good to me 👍